dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DubboProtocol.java

```diff
@@ -702,6 +728,8 @@ public void destroy() {
             }
         }
         referenceClientMap.clear();
+        
+        stubServiceToLatch.clear();
```

Comment:
I think we should take destroy of stubServiceToLatch on destroy of stub proxy too.

There might have lots of proxy invokers (reference config) create and destroy during the lifetime of DubboProtocol.

Reply:
Consider the complexities of `latch.await()`, Can `stubServiceToLatch` be replaced with this similar structure, `Map<String,Boolean> stubService2Flag`.When the `connect` event occurs and the `stub invoker` is not found, execute `stubService2Flag.put(serviceKey, true)`.
When the stub service exposure is completed, check `if(stubService2Flag.get(serviceKey))`, then call `received(channel, invocation);`

Reply:
`Map<String,Boolean> stubService2Flag` is still a variable that needs to be taken care of in case of a memory leak. I think a detailed log giving full information of possible causes would be fine.

Reply:
ok, then there is no need to wait(`stubServiceToLatch`) or delay notification(`stubService2Flag`). If the `stub invoker `cannot be obtained, just print the detailed log.